Bobjayafam/politico
# #163637964 implement endpoint for user login

#### What does this PR do?
Implement a user login endpoint which allows a user to login to the application
#### Description of Task to be completed?
- add a user login method to the user controller to handle the logic
- add a middleware to valid user login data
- write tests for the endpoint
#### How should this be manually tested?
- clone the GitHub repo
- checkout branch ``ft-user-login-163637964``
- run npm install
- run npm run dev
- on POSTMAN, send a POST request to ``/api/v1/auth/login``. send a request body with an already created user email and password. The response should contain a token, user object and a status code of 200
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/163637964
#### Screenshots (if appropriate)

![login-user](https://user-images.githubusercontent.com/19153324/52143267-55826880-265b-11e9-93a8-09ed7de7ec88.png)
